frontend/src/components/Field/Rating/Rating.tsx

```diff
@@ -0,0 +1,186 @@
+import { Fragment, useEffect, useMemo, useState } from 'react'
+import {
+  Box,
+  forwardRef,
+  Grid,
+  Stack,
+  Text,
+  useBreakpointValue,
+  Wrap,
+  WrapItem,
+} from '@chakra-ui/react'
+
+import { FieldColorScheme } from '~theme/foundations/colours'
+
+import { RatingOption } from './RatingOption'
+
+export interface RatingProps {
+  /**
+   * The `name` attribute forwarded to each rating `radio` element
+   */
+  name: string
+  /**
+   * Number of rating options to render.
+   */
+  numberOfRatings: number
+  /**
+   * Color scheme of the component to render. Defaults to `primary`.
+   */
+  colorScheme?: FieldColorScheme
+  /**
+   * Function called once a rating is selected.
+   * @param newRating the value of the checked radio
+   */
+  onChange?: (newRating: number | undefined) => void
+  /**
+   * Number of rating components to show per row when unable to display all
+   * components on a single line. Defaults to `5`.
+   */
+  wrapComponentsPerRow?: number
+  /**
+   * The value of the rating to be `checked` initially.
+   */
+  defaultValue?: number
+
+  /**
+   * Variant of rating field to render
+   */
+  variant: 'Heart' | 'Star' | 'Number'
```

Comment:
we can consider making this lowercase so that it's not coupled to our formsg enums